src/main.ts

```diff
@@ -1,101 +1,111 @@
+        if (!servicePrincipalId || !servicePrincipalKey || !tenantId || !subscriptionId) {
+            throw new Error("Not all values are present in the creds object. Ensure clientId, clientSecret, tenantId and subscriptionId are supplied.");
+        }
+
+        // Attempting Az cli login
+        if (environment.toLowerCase() == "azurestack") {
+            if (!resourceManagerEndpointUrl) {
+                throw new Error("resourceManagerEndpointUrl is a required parameter when environment is defined.");
+            }
+            console.log(`Unregistering cloud: "${environment}" first if it exists`);
+            try {
+                await executeAzCliCommand(`cloud set -n AzureCloud`, true);
+                await executeAzCliCommand(`cloud unregister -n "${environment}"`, false);
+            } catch (error) {
+                console.log(`Ignore cloud not registered error: "${error}"`);
+            }
+            console.log(`Registering cloud: "${environment}" with ARM endpoint: "${resourceManagerEndpointUrl}"`);
+            try {
+                let baseUri = resourceManagerEndpointUrl;
+                if (baseUri.endsWith('/')) {
+                    baseUri = baseUri.substring(0, baseUri.length-1); // need to remove trailing / from resourceManagerEndpointUrl to correctly derive suffixes below
+                }
+                let suffixKeyvault = ".vault" + baseUri.substring(baseUri.indexOf('.')); // keyvault suffix starts with .
+                let suffixStorage = baseUri.substring(baseUri.indexOf('.')+1); // storage suffix starts without .
+                let profileVersion = "2019-03-01-hybrid";
+                await executeAzCliCommand(`cloud register -n "${environment}" --endpoint-resource-manager "${resourceManagerEndpointUrl}" --suffix-keyvault-dns "${suffixKeyvault}" --suffix-storage-endpoint "${suffixStorage}" --profile "${profileVersion}"`, false);
+            } catch (error) {
+                core.error(`Error while trying to register cloud "${environment}": "${error}"`);
+            }
+            await executeAzCliCommand(`cloud set -n "${environment}"`, false);
+            console.log(`Done registering cloud: "${environment}"`);
+        }
+           
+        //setting context to Azure Gov Cloud
+        if (environment.toLowerCase() == AzureCloudName.AzureUSGovernment|| environment.toLowerCase() ==AzureCloudName.AzureChinaCloud || environment.toLowerCase() ==AzureCloudName.AzureGermanCloud  || environment.toLowerCase() ==AzureCloudName.AzureCloud  ) {
+            await executeAzCliCommand(`cloud set --name "${environment}"`, true);       
+        } 
+    
+        isAzCLISuccess = true;
+        if (enableAzPSSession) {
+            // Attempting Az PS login
+            console.log(`Running Azure PS Login`);
+            const spnlogin: ServicePrincipalLogin = new ServicePrincipalLogin(servicePrincipalId, servicePrincipalKey, tenantId, subscriptionId, environment, resourceManagerEndpointUrl);
```

Comment:
please add changes to powershell login for sovereign cloud.

Reply:
The Connect-AzAccount cmdlet will have to be updated to include Environment parameter as: 
```powershell
Connect-AzAccount -Environment {environmentName}
```

Reply:
@asranja yes, i missed it. just discussed with @zainuvk. "-Environment" will handle azure powershell login.